packages/store/src/workspace/page.ts

```diff
@@ -14,21 +14,30 @@ import type { IdGenerator } from '../utils/id-generator.js';
 import { Signal } from '../utils/signal.js';
 import {
   assertValidChildren,
-  initSysProps,
   syncBlockProps,
   trySyncTextProp,
   toBlockProps,
   matchFlavours,
 } from '../utils/utils.js';
 import type { PageMeta, Workspace } from './workspace.js';
+import type { EnhancedYMap } from '../utils/yjs/proxy.js';
+import { createYMap } from '../utils/yjs/index.js';
 
-export type YBlock = Y.Map<unknown>;
-export type YBlocks = Y.Map<YBlock>;
+export interface YBlockData {
+  'sys:id': string;
+  'sys:flavour': string;
+  'sys:children': Y.Array<string>;
+
+  [key: string]: unknown;
+}
+
+export type YBlock = EnhancedYMap<YBlockData>;
+export type YBlocks = EnhancedYMap<Record<string, YBlock>>;
 
 /** JSON-serializable properties of a block */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type BlockProps = Record<string, any> & {
-  id: string;
+  // id: string;
```

Comment:
I think we shouldn't let use to pass an id though props. because it is random generated